lib/seahorn/BvOpSem2.cc

```diff
@@ -1240,7 +1240,11 @@ class OpSemVisitor : public InstVisitor<OpSemVisitor>, OpSemVisitorBase {
       Expr isNoOverflow =
           m_ctx.alu().IsUaddNoOverflow(op0, op1, ty->getScalarSizeInBits());
       assert(addRes && isNoOverflow);
-      Expr maxVal = m_ctx.alu().si(~0UL, ty->getScalarSizeInBits());
+      mpz_class maxValZ;
+      for (unsigned i = 0; i < ty->getScalarSizeInBits(); ++i) {
+        maxValZ.setbit(i);
+      }
```

Comment:
why is this better than `maxValZ = 2^(ty->getScalarSizeInBits()) - 1`?

Reply:
The question is what space to do the computation at. This way, the computation is in infinite precision space, so it is guaranteed to be precise. If we do it in machine arithmetic, it might underflow depending on the architecture (i.e., size of `unsigned`) and requested number of bits. This was the problem with the previous solution that caused some test to fail.